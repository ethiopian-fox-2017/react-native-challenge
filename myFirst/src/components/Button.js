import React, { Component } from 'react';

import {
  Text,
  TouchableOpacity
} from 'react-native';
import styles from '../assets/index.js'


class Button extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
    }
  }

  render() {
    return (
      <TouchableOpacity style={styles.button} onPress={() => {}}>
        <Text>Display Data</Text>
      </TouchableOpacity>
    )
  }
}

export default Button;
