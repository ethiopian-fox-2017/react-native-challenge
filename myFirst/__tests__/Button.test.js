import 'react-native';
import React from 'react';
import Button from '../src/components/Button';
import { shallow } from 'enzyme';
// Note: test renderer must be required after react-native.


test('renders correctly', () => {
  const tree = shallow(
    <Button />
  );
  expect(tree).toMatchSnapshot();
});
