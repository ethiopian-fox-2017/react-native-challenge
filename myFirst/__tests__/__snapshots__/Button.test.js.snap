// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
        Object {
            "backgroundColor": "white",
            "borderColor": "teal",
            "borderRadius": 10,
            "borderWidth": 2,
            "margin": 10,
            "padding": 10,
          }
    }
>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Display Data
    </Text>
</TouchableOpacity>,
  "nodes": Array [
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
            Object {
                  "backgroundColor": "white",
                  "borderColor": "teal",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "margin": 10,
                  "padding": 10,
                }
      }
>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
      >
            Display Data
      </Text>
</TouchableOpacity>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "data": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
                    Object {
                              "backgroundColor": "white",
                              "borderColor": "teal",
                              "borderRadius": 10,
                              "borderWidth": 2,
                              "margin": 10,
                              "padding": 10,
                            }
          }
>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Display Data
          </Text>
</TouchableOpacity>,
        "_debugID": 2,
        "_renderedOutput": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
                    Object {
                              "backgroundColor": "white",
                              "borderColor": "teal",
                              "borderRadius": 10,
                              "borderWidth": 2,
                              "margin": 10,
                              "padding": 10,
                            }
          }
>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Display Data
          </Text>
</TouchableOpacity>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button />,
}
`;
